src/main/java/com/library/model/mapper/AuthorMapper.java

```diff
@@ -0,0 +1,74 @@
+package com.library.model.mapper;
+
+import com.library.model.dto.AuthorDto;
+import com.library.model.dto.AuthorNameDto;
+import com.library.model.entity.Author;
+import org.modelmapper.ModelMapper;
+import org.modelmapper.PropertyMap;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+
+
+@Component
+public class AuthorMapper {
+
+    private final ModelMapper modelMapper;
+
+    public AuthorMapper(ModelMapper modelMapper) {
+        this.modelMapper = modelMapper;
+
+        modelMapper.createTypeMap(Author.class, AuthorDto.class)
+                .addMappings(new PropertyMap<Author, AuthorDto>() {
+                    @Override
+                    protected void configure() {
+                        using(context -> generatedFullName(
+                                ((Author)context.getSource()).getName(),
+                                ((Author)context.getSource()).getSurname()))
+                                .map(source, destination.getFullName());
+                    }
+                });
+
+        modelMapper.createTypeMap(Author.class, AuthorNameDto.class)
+                .addMappings(new PropertyMap<Author, AuthorNameDto>() {
+                    @Override
+                    protected void configure() {
+                        using(context -> generatedFullName(
```

Comment:
Duplicated code.
Try to remove duplicate.

Reply:
See in commit.